### DISCUSSION TITLE
is there any plan to support multithreading?

### DISCUSSION THREAD

**Discussion by ari-stern** [slingdata-io/sling-cli#30] [Ideas]
it would be nice to be able to refresh multiple tables at once

**Comment by flarco** [comment]
Hi, actually this was a thought earlier but was put off because of the potential stress that it can cause on the source database system. You can bring a system down if you select multiple sizable, whole tables at once. I believe ELT platforms today (such as Stitch, and Fivetran) do one table at a time as well, probably for that same reason.

But, with that fact that Sling is built in Go, multi threading can be added with relative ease. So I think that this could be a configurable parameter with [replications](https://docs.slingdata.io/sling-cli/replication) mode. Let me think a little more about this. We could have some top level parameter like `threads`, with the default number being `1` if not specified.

**Reply by ari-stern** [reply]
cool thank you!

**Comment by jmesterh** [comment]
I agree with the potential stress issue, but it would be very nice to have the _option_ to multithread at the table level, given appropriate warning in the documentation.

**Reply by flarco** [reply]
Will revisit this soon.

**Comment by flarco** [comment]
Opened https://github.com/slingdata-io/sling-cli/issues/117.
